routes/personal/personal.controller.js

```diff
@@ -79,3 +87,25 @@ const postMaritalStatus = (req, res) => {
 
   return res.redirect(req.body.redirect)
 }
+
+const postResidence = (req, res) => {
+  const errors = validationResult(req)
+
+  let residence = req.body.residence || null
+  req.session.personal = {
+    residence: residence,
+  }
```

Comment:
we don't want to assign until we know things are successful, so let's pop this down after the `if(!errors.isEmpty())` check.
Also we can do away with the `let` up there since we're not manipulating the data into any shape. 
We also don't want to override the whole object, so I think we can change lines 94-97 to one line which is just `req.session.personal.residence = req.body.residence`
We updated how we're doing this recently, so if you look at the latest version of master you can see maritalStatus has been updated like this as well

Reply:
going to remove this actually. Had a chat with @pcraig3 and we aren't actually going to be saving this value anywhere, just using it to redirect to corresponding pages.

Reply:
Yeah, axe it.